File: e2e-tests/src/test/java/e2e/models/ProductPayload.java
package e2e.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProductPayload {
    private Long barcodeId;
    private String shortName;
    private String description;
    private BigDecimal price;
    private int quantity;
    private String addedAtWarehouse;
    private boolean isFoodstuff;
}

File: e2e-tests/src/main/kotlin/models/ProductPayload.kt
package models

import com.fasterxml.jackson.annotation.JsonProperty
import java.math.BigDecimal

data class ProductPayload(
    var barcodeId: Long? = null,
    var shortName: String? = null,
    var description: String? = null,
    var price: BigDecimal = BigDecimal.ZERO,
    var quantity: Int = 0,
    var addedAtTariffs: String? = null,
    @field:JsonProperty("isFoodstuff")
    @get:JsonProperty("isFoodstuff")
    var isFoodstuff: Boolean = false
)

